shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -95,6 +95,24 @@ public Counter call() throws Exception {
     }
 
     @Override
+    public long getCounterValue(String name) {
+        Exceptions.checkNotNullOrEmpty(name, "name");
+        String counterName = name + ".Counter";
+        try {
+            Counter counter = countersCache.get(counterName, new Callable<Counter>() {
+                @Override
+                public Counter call() throws Exception {
+                    return underlying.createCounter(counterName);
```

Comment:
I think my comment wasn't clear, so let me try again. `getCounterValue` is supposed to return a counter value and getters typically do not have side-effects. I understand that the `get` call in the cache enables the behavior implemented with the `Callable`, but I'm not sure we should be making use of it here. Could you convince me that the expected behavior is to create a counter with zero value or populate it with the lambda that @shiveshr suggests?